js/src/lib/interfaces/Libraries.ts

```diff
@@ -106,11 +106,41 @@ const asteroid = (model: ModelData) =>
 
 model = BaseModel.from_pretrained("${model.id}")`;
 
-const diffusers = (model: ModelData) =>
-	`from diffusers import DiffusionPipeline
+const diffusers_default = (model: ModelData) =>
+  `from diffusers import DiffusionPipeline
 
 pipeline = DiffusionPipeline.from_pretrained("${model.id}")`;
 
+const diffusers_controlnet = (model: ModelData) =>
+  `from diffusers import StableDiffusionControlNetPipeline, ControlNetModel
+
+controlnet = ControlNetModel.from_pretrained("${model.id}")
+pipeline = StableDiffusionControlNetPipeline.from_pretrained("${model.tags.base_model}", controlnet=controlnet)`;
```

Comment:
`model.tags` is just an array of strings: https://github.com/huggingface/hub-docs/blob/8be1210dc51f1cdfa76703a9ec8dd4c76c0ff728/js/src/lib/interfaces/Types.ts#L667

Not sure if `base_model` is part of the `config`?

Reply:
`base_model` is not part of the `config` (which comes from `config.json`), but from the model card metadata (`README.md`), which should be under `model.cardData?.base_model`. Pointers

* Example model card https://huggingface.co/hf-internal-testing/tiny-controlnet/blob/main/README.md 
* Related PR: https://github.com/huggingface/api-inference-community/pull/223 

For confirmation

```python
from huggingface_hub import model_info
model_info("hf-internal-testing/tiny-controlnet")
```
gives
```
ModelInfo: { 
  {
   'author': 'hf-internal-testing',
   'cardData': {'base_model': 'hf-internal-testing/tiny-stable-diffusion-pipe',
                'license': 'openrail',
                'tags': ['art', 'controlnet', 'stable-diffusion', 'image-to-image']},
   'config': {},
   'disabled': False,
   ...
}
```

so indeed it's under `model.cardData?.base_model` by default

Reply:
@osanseviero yes, but I guess the type interface needs to include it here: https://github.com/huggingface/hub-docs/blob/8be1210dc51f1cdfa76703a9ec8dd4c76c0ff728/js/src/lib/interfaces/Types.ts#L717-L721

Is it enough to add it there to have it populated from the repo?

Reply:
@patrickvonplaten sorry, I pushed a commit just to check if the tests pass. They now do, but I'm still not sure if `base_model` will be automatically populated after adding it to the type interface.

Reply:
`base_model` is already populated, the types are only partial but the data was already in there

Reply:
Awesome, thanks for the clarification!